webhooks-extension/cmd/interceptor/interceptor.go

```diff
@@ -113,8 +114,19 @@ func main() {
 				wantedEvent := request.Header.Get("Wext-Incoming-Event")
 				foundEvent := request.Header.Get("X-Github-Event")
 				if wantedEvent == foundEvent { // Wanted GitHub event type provided AND repository URL matches so all is well
-					validationPassed = true
-					log.Printf("[%s] Validation PASS (repository URL, secret payload, event type checked)", foundTriggerName)
+					wantedActions := request.Header["Wext-Incoming-Actions"]
+					if len(wantedActions) == 0 {  
+						validationPassed = true
+						log.Printf("[%s] Validation PASS (repository URL, secret payload, event type checked)", foundTriggerName)
+					} else {
+						actions := strings.Split(wantedActions[0], ",")
+						for _, action := range actions {
+							if action == result.Action {
+								validationPassed = true
+								log.Printf("[%s] Validation PASS (repository URL, secret payload, event type, action:%s checked)", foundTriggerName, action)
+							}
+						}
```

Comment:
I think that this change surfaces a bug in the existing interceptor code. When `validationPassed` is `false`, it is never written as an error. On line 155 (https://github.com/tektoncd/experimental/pull/354/files#diff-4e40b3eebcbb16692d761b108f341566R155) we should have an `else` statement to catch the case when `validationPassed` is false. For example:
```go
if validationPassed {
	log.Printf("[%s] Validation PASS so writing response", foundTriggerName)
	_, err := writer.Write(payload)
	if err != nil {
		log.Printf("[%s] Failed to write response for Github event ID: %s. Error: %s", foundTriggerName, id, err.Error())
		http.Error(writer, fmt.Sprint(err), http.StatusInternalServerError)
		return
	}
} else {
	http.Error(writer, "Validation failed", http.StatusExpectationFailed)
}
```

Reply:
@ncskier Good catch!!  I fix it.  Thanks!